src/poliastro/plotting/porkchop.py

```diff
@@ -134,120 +123,146 @@ def porkchop(
-    >>> from poliastro.util import time_range
-    >>> launch_span = time_range("2005-04-30", end="2005-10-07")
-    >>> arrival_span = time_range("2005-11-16", end="2006-12-21")
-    >>> dv_launch, dev_dpt, c3dpt, c3arr, tof = porkchop(Earth, Mars, launch_span, arrival_span)
     """
 
-    dv_launch, dv_arrival, c3_launch, c3_arrival, tof = targetting_vec(
+    def __init__(
+        self,
         departure_body,
         target_body,
-        launch_span[np.newaxis, :],
-        arrival_span[:, np.newaxis],
-    )
-
-    # Start drawing porkchop
-
-    if ax is None:
-        fig, ax = plt.subplots(figsize=(15, 15))
-    else:
-        fig = ax.figure
-
-    c3_levels = np.linspace(0, max_c3.to(u.km ** 2 / u.s ** 2).value, 30)
-
-    c = ax.contourf(
-        [D.to_datetime() for D in launch_span],
-        [A.to_datetime() for A in arrival_span],
-        c3_launch,
-        c3_levels,
-    )
-
-    line = ax.contour(
-        [D.to_datetime() for D in launch_span],
-        [A.to_datetime() for A in arrival_span],
-        c3_launch,
-        c3_levels,
-        colors="black",
-        linestyles="solid",
-    )
-
-    cbar = fig.colorbar(c)
-    cbar.set_label("km2 / s2")
-    ax.clabel(line, inline=1, fmt="%1.1f", colors="k", fontsize=10)
-
-    if tfl:
-
-        time_levels = np.linspace(100, 500, 5)
-
-        tfl_contour = ax.contour(
-            [D.to_datetime() for D in launch_span],
-            [A.to_datetime() for A in arrival_span],
-            tof,
-            time_levels,
-            colors="red",
-            linestyles="dashed",
-            linewidths=3.5,
+        launch_span,
+        arrival_span,
+        ax=None,
+        tfl=True,
+        vhp=True,
+        max_c3=45.0 * u.km ** 2 / u.s ** 2,
+        max_vhp=5 * u.km / u.s,
+    ):
+        self.departure_body = departure_body
+        self.target_body = target_body
+        self.launch_span = launch_span
+        self.arrival_span = arrival_span
+        self.ax = ax
+        self.tfl = tfl
+        self.vhp = vhp
+        self.max_c3 = max_c3
+        self.max_vhp = max_vhp
+
+    def porkchop(self):
+        """Plots porkchop between two bodies.
+        Returns
+        -------
+        dv_launch: np.ndarray
+            Launch delta v
+        dv_arrival: np.ndarray
+            Arrival delta v
+        c3_launch: np.ndarray
+            Characteristic launch energy
+        c3_arrrival: np.ndarray
+            Characteristic arrival energy
+        tof: np.ndarray
+            Time of flight for each transfer
+        Example
+        -------
+        >>> from poliastro.plotting.porkchop import porkchop
+        >>> from poliastro.bodies import Earth, Mars
+        >>> from poliastro.util import time_range
+        >>> launch_span = time_range("2005-04-30", end="2005-10-07")
+        >>> arrival_span = time_range("2005-11-16", end="2006-12-21")
+        >>> dv_launch, dev_dpt, c3dpt, c3arr, tof = porkchop(Earth, Mars, launch_span, arrival_span)
```

Comment:
The `Example` section should be adapted to new API namespace. No longer `porkchop` but `PorkchopPlotter` :wink:

Reply:
Sure thing!! That passed under my nose:wink: